### PR TITLE
[codemod] Rename `shouldDisableTime` prop

### PR DESCRIPTION
**Pull request by alexfauquette**
Part of #6997

### CODE REVIEW

File: packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+
+<div>
+  <DateTimePicker shouldDisableTime={(timeValue, view) => view === 'hours' && timeValue < 12} />
```
Comment (LukasTy) [inline packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js:4]: We could also introduce a test case with function as a variable just for the sake wider coverage. 🙈 

```diff
diff --git a/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js b/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js
index 898616043..bf32e31b8 100644
--- a/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js
+++ b/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js
@@ -1,5 +1,9 @@
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+
+const shouldDisable = (timeValue, view) => view === 'hours' && timeValue < 12;
 
 <div>
   <DateTimePicker shouldDisableTime={(timeValue, view) => view === 'hours' && timeValue < 12} />
+  <TimePicker shouldDisableTime={shouldDisable} />
 </div>;
diff --git a/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/expected-props.spec.js b/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/expected-props.spec.js
index 784d823db..f3e13aeaf 100644
--- a/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/expected-props.spec.js
+++ b/packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/expected-props.spec.js
@@ -1,5 +1,9 @@
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+
+const shouldDisable = (timeValue, view) => view === 'hours' && timeValue < 12;
 
 <div>
   <DateTimePicker shouldDisableClock={(timeValue, view) => view === 'hours' && timeValue < 12} />
+  <TimePicker shouldDisableClock={shouldDisable} />
 </div>;

```

**Review by LukasTy — APPROVED** [review]
Looks great! 💯 
Just a general question: do you know if such thing is feasible with `jscodeshift`:
Having:
```tsx
const props = {
  shouldDisableTime: (timeValue, view) => view === 'hours' && timeValue < 12,
};

<TimePicker {...props} />
```
transform it into:
```tsx
const props = {
  shouldDisableClock: (timeValue, view) => view === 'hours' && timeValue < 12,
};

<TimePicker {...props} />
```

Because our current codemods will only cover users directly using pickers from our package as well as specifying props directly on the component, but I fear that in the real world we might have quite a lot of users doing the following:
- having a wrapper component on top of ours with additional logic/defaults
- spreading the props from some other object onto the picker component

File: packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js (hunk 1 above)
Comment (alexfauquette) [reply packages/x-codemod/src/v6.0.0/pickers/rename-should-disable-time/actual-props.spec.js:7]: If you feel safer with it, I add it with pleasure :)

**Comment by alexfauquette** [comment]
@LukasTy I don't think it's feasible with `jscodeshift` because we use AST which represent the code as tokens, but do not set any logic between them.


For example here is a simple code and it's associated AST in JSON (I removed most of the useless parts)
```jsx
const prop = { label: "Name" }
<A {...props} />
```

```json
{
  "program": {
    "type": "Program",
    "body": [
      {
        "type": "VariableDeclaration",
        "declarations": [
          {
            "type": "VariableDeclarator",
            "id": {
              "type": "Identifier",
              "name": "props"
            },
            "init": {
              "type": "ObjectExpression",
              "properties": [
                {
                  "type": "Property",
                  "key": {
                    "type": "Identifier",
                    "name": "label"
                  },
                  "computed": false,
                  "value": {
                    "type": "Literal",
                    "value": "Nale",
                    "raw": "'Name'"
                  },
                  "kind": "init",
                  "method": false,
                  "shorthand": false
                }
              ]
            }
          }
        ],
        "kind": "const"
      },
      {
        "type": "ExpressionStatement",
        "expression": {
          "type": "JSXElement",
          "openingElement": {
            "type": "JSXOpeningElement",
            "name": {
              "type": "JSXIdentifier",
              "name": "A"
            },
            "selfClosing": true,
            "attributes": [
              {
                "type": "JSXSpreadAttribute",
                "argument": {
                  "type": "Identifier",
                  "name": "props"
                }
              }
            ]
          },
          "children": [],
          "closingElement": null
        }
      }
    ]
  }
}
```

The tree is made of two parts:
- `type: "VariableDeclaration"` which declares the definition of `props`
- `type: "ExpressionStatement"` that defines the components

The only way to get the link between them is to notice that the first one defines an object of type `"Identifier"` with the name `"props"`  which has the same name as the one used in the spread operator.

As you can imagine, this can not scale since you would have to consider all the possible ways to edit an object. Same if objects are imported (we only parse one file at a time)

**Comment by LukasTy** [comment]
@alexfauquette Thank you for a detailed answer. 🙏

**Pull request merged**